File: plugins/node/template/addons/s3/src/config/storage.ts
import { z } from "zod";

const storageSchema = z.object({
  S3_BUCKET: z.string().min(1, "S3_BUCKET is required"),
  S3_REGION: z.string().min(1).default("us-east-1"),
  S3_ENDPOINT: z.string().optional(),
  AWS_ACCESS_KEY_ID: z.string().min(1, "AWS_ACCESS_KEY_ID is required"),
  AWS_SECRET_ACCESS_KEY: z.string().min(1, "AWS_SECRET_ACCESS_KEY is required"),
});

export const storageEnv = storageSchema.parse(process.env);

File: plugins/python/template/addons/s3/app/core/storage.py
from typing import Optional

from pydantic_settings import BaseSettings, SettingsConfigDict


class StorageSettings(BaseSettings):
    S3_BUCKET: str = ""
    S3_REGION: str = "us-east-1"
    S3_ENDPOINT: Optional[str] = None
    AWS_ACCESS_KEY_ID: str = ""
    AWS_SECRET_ACCESS_KEY: str = ""

    model_config = SettingsConfigDict(env_file=".env", case_sensitive=True, extra="ignore")


storage_settings = StorageSettings()
